### PR TITLE
Feature 3 authentication support

### LINKED ISSUE #3: Authentication support
As a user I'd like to have authentication using Keycloak

### PR DESCRIPTION
**Pull request by EVP1989**
Resolves #3

### CODE REVIEW

File: src/components/layout-components/header.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        }}
+      />
+    )
+  }
+
+  return (
+    <Stack direction="row-reverse">
+      <Stack direction="row">
+        <Stack sx={{ marginRight: 2 }}>
+          <Avatar>img</Avatar>
```
Comment (villejuutila) [inline src/components/layout-components/header.tsx:35]: Avatar is a self closing component  so change this to <Avatar />.

**Review by villejuutila — CHANGES_REQUESTED** [review]

File: src/components/layouts/basic-layout.tsx [hunk 2]
```diff
@@ -1,28 +1,31 @@
+import Header from "../layout-components/header";
 import React from "react";
 
 /**
  * Component props
  */
 export interface Props {
-  children: React.ReactNode
+  children: React.ReactNode;
```
[the remark this answers is not in the document]
Comment (EVP1989) [reply src/components/layouts/basic-layout.tsx:8]: True, fixed

File: src/components/layout-components/header.tsx (hunk 1 above)
Comment (EVP1989) [reply src/components/layout-components/header.tsx:35]: Fixed

File: src/components/providers/authentication-provider.tsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
+import { useCallback, useEffect } from "react";
+import Keycloak from "keycloak-js";
+import { useAtom, useSetAtom } from "jotai";
+import { authAtom, userProfileAtom } from "../../atoms/auth";
+import config from "../../app/config";
+
+type Props = {
```
[the remark this answers is not in the document]
Comment (EVP1989) [reply src/components/providers/authentication-provider.tsx:7]: Fixed

File: src/App.tsx [hunk 4]
```diff
@@ -24,11 +25,13 @@ const router = createBrowserRouter([
 const App = () => (
   <div className="App">
     <ErrorHandler>
-      <BasicLayout>
+      <AuthenticationProvider>
+        <BasicLayout>
         <RouterProvider router={router} />
       </BasicLayout>
```
Comment (villejuutila) [inline src/App.tsx:31]:
```suggestion
          <RouterProvider router={router} />
        </BasicLayout>
```

**Review by villejuutila — APPROVED** [review]

**Pull request merged**